### ISSUE TITLE
[Research] Training data curation

### ISSUE THREAD

**Issue by shash42** [OpenForecaster/scaling-forecasting-training#3]
In our [paper](https://www.alphaxiv.org/abs/2512.25070), Table 1 shows effects of filtering. Without leakage filtering, our model deteriorated. 

This shows the high potential impact of data curation for forecasting. We only took the most obvious first steps, filtering questions using another model.

There is a lot of scope in designing better question generation and filtering pipelines. For example, later in our manual review (Section 6 of paper) of the test set we found a large number of resolution dates were wrong. These issues still affect our training data, and our performance gains are despite them. It is quite likely fixing these issues will lead to even more performance gains.

How to balance the cost-benefit tradeoff here which allows question generation at scale while still improving quality? One option could be to use search agents to set accurate resolution dates. For our test set, we used Grok 4 search, but this can be too expensive at training scale. Cheaper solutions might involve distillation, better prompts (e.g. few-shot examples and guidelines), or even rules. 

Resolution, and wordings of forecasting questions can be a subtle, complex issue. But its one that AI agents can solve, and probably easier than forecasting itself (think generator-verifier gap). There's also a good amount of data, including edge-cases, from prediction markets to test on.
